llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -560,11 +560,12 @@ bool AsmPrinter::doInitialization(Module &M) {
   }
 
   if (MAI->doesSupportDebugInformation()) {
-    bool EmitCodeView = M.getCodeViewFlag();
-    if (EmitCodeView &&
-        (TM.getTargetTriple().isOSWindows() || TM.getTargetTriple().isUEFI()))
+    // On Windows targets, emit minimal CodeView compiler info even when debug
+    // info is disabled.
+    if ((TM.getTargetTriple().isOSWindows() || TM.getTargetTriple().isUEFI()) &&
+        M.getNamedMetadata("llvm.dbg.cu"))
       Handlers.push_back(std::make_unique<CodeViewDebug>(this));
-    if (!EmitCodeView || M.getDwarfVersion()) {
+    if (M.getDwarfVersion() || !M.getCodeViewFlag()) {
```

Comment:
Just being picky here, but can you please leave the condition in the same order to ease downstream integrations?